components/script/indexed_db.rs

```diff
@@ -74,6 +74,24 @@ pub fn key_type_to_jsval(
                 values.push(Heap::boxed(value.get()));
             }
             values.safe_to_jsval(cx, result, can_gc);
+
+            // rooted!(in(*cx) let mut arr = unsafe { js::jsapi::NewArrayObject1(*cx, a.len() as _) });
```

Comment:
@jdm for fixed length cases I can use ValueArray::new from mozjs, but for dynamic lengths I can’t. I’ve left this proposed commented out , could you take a look?